client/src/helpers/fhir/builder/ServiceRequestBuilder.ts

```diff
@@ -153,17 +143,67 @@ export class ServiceRequestBuilder {
       }
       return this;
     }
+  
+    public withStatus(status?: string) {
+      if (status) {
+        this.serviceRequest.status = status
+      }
+      return this;
+    }
+  
+    public withNoteStatus(note?: string, practitionerId?: string) {
+      if (note && note.length > 0 && practitionerId) {
+        this.serviceRequest.note = updateNoteStatus({
+          authorReference: {
+            reference: `Practitioner/${practitionerId}`,
+          },
+          text: note,
+          time: new Date().toISOString(),
+        }, this.serviceRequest.note);
+      }
+      return this;
+    }
+  
+    public withProcedureDirectedBy(practitionerRoleId?: string) {
+      if (practitionerRoleId) {
+        const ext = getExtension(this.serviceRequest, ExtensionUrls.ProcedureDirectedBy);
+        if (ext) {
+          ext.valueReference = {
+            reference: `PractitionerRole/${practitionerRoleId}`,
+          };
+        } else {
```

Comment:
I think the else clause is not needed.

Reply:
It's required, a ServiceRequest doesn't contain all extensions, we update or add them when needed.